jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -198,22 +198,28 @@ public long length()
     /**
      * Creates a new input stream to the resource.
      *
-     * @return an input stream to the resource
-     * @throws IOException if unable to open the input stream
+     * @return an input stream to the resource or null if one is not available.
+     * @throws IOException if there is a problem opening the input stream
      */
     public InputStream newInputStream() throws IOException
     {
-        return Files.newInputStream(getPath(), StandardOpenOption.READ);
+        Path path = getPath();
+        if (path == null)
+            return null;
+        return Files.newInputStream(path, StandardOpenOption.READ);
     }
 
     /**
      * Readable ByteChannel for the resource.
      *
-     * @return an readable bytechannel to the resource or null if one is not available.
+     * @return a readable {@link java.nio.channels.ByteChannel} to the resource or null if one is not available.
      * @throws IOException if unable to open the readable bytechannel for the resource.
      */
     public ReadableByteChannel newReadableByteChannel() throws IOException
```

Comment:
This should probably be a `SeekableByteChannel` instead.
(Would be more generally useful in the static byte-range support as well)

Reply:
The `Files.newByteChannel()` already returns `SeekableByteChannel`

Reply:
probably... I'll do that in another PR.

Reply:
Opened as PR #8772